pkg/registry/generic/rest/streamer.go

```diff
@@ -56,14 +58,24 @@ func (s *LocationStreamer) InputStream(apiVersion, acceptHeader string) (stream
 	if err != nil {
 		return
 	}
-	contentType = s.ContentType
-	if len(contentType) == 0 {
-		contentType = resp.Header.Get("Content-Type")
-		if len(contentType) > 0 {
-			contentType = strings.TrimSpace(strings.SplitN(contentType, ";", 2)[0])
+
+	if resp.StatusCode >= 200 && resp.StatusCode < 300 {
+		contentType = s.ContentType
+		if len(contentType) == 0 {
+			contentType = resp.Header.Get("Content-Type")
+			if len(contentType) > 0 {
+				contentType = strings.TrimSpace(strings.SplitN(contentType, ";", 2)[0])
+			}
 		}
+		flush = s.Flush
+		stream = resp.Body
+		return
+	}
+
+	bodyBytes, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, false, "", err
 	}
-	flush = s.Flush
-	stream = resp.Body
-	return
+	bodyText := string(bodyBytes)
+	return nil, false, "", fmt.Errorf("%s", bodyText)
```

Comment:
Comments:
* Tests.
* Move the error handling above the success case-- it's not obvious that this is the error case.
* Is any code other than 200 valid?
* Return a StatusError (IIRC that's the type name)

If the server in fact is going to stream and we incorrectly think it won't, that ReadAll is going to block for a very long time, which is making me a little nervous.